yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala

```diff
@@ -337,5 +339,82 @@ object YarnSparkHadoopUtil {
       conf.getInt("spark.executor.instances", targetNumExecutors)
     }
   }
+
+  /**
+   * Obtains token for the Hive metastore. Exceptions are caught and logged
+   */
+  def obtainTokenForHiveMetastore(
+      sparkConf: SparkConf,
+      conf: Configuration): Option[Token[DelegationTokenIdentifier]] = {
+    try {
+      obtainTokenForHiveMetastoreInner(sparkConf, conf,
+        UserGroupInformation.getCurrentUser().getUserName)
+    } catch {
+      case e: java.lang.NoSuchMethodException =>
+        logInfo("Hive Method not found " + e)
+        None
+      case e: java.lang.ClassNotFoundException =>
+        logInfo("Hive Class not found " + e)
+        None
+      case e: java.lang.NoClassDefFoundError =>
+        logDebug("Hive Class not found: " + e)
```

Comment:
This may not be class not found exception, but it could be an initialization exception, where the HiveConf or Hadoop Conf is not available during class contruction